docs: adding algokit template documentation (#300)
algorandfoundation/algokit-cli: docs/features/init.md

? Continue anyway? Yes
```

> If you want to create a community template, you can use the [beaker template](https://github.com/algorandfoundation/algokit-cli/blob/main/docs/tutorials/algokit-template.md#creating-algokit-templates) and [Copier documentation](https://copier.readthedocs.io/en/stable/) as a starting point.

Reviewer: ```suggestion
If you want to create a community template, you can use [our guidelines on template building](https://github.com/algorandfoundation/algokit-cli/blob/main/docs/tutorials/algokit-template.md#creating-algokit-templates) and [Copier documentation](https://copier.readthedocs.io/en/stable/) as a starting point.
```
